sketch/edge_length.py

```diff
@@ -0,0 +1,286 @@
+            case 1:
+                vertex_0.co = self.starting_positions[index][0]
+                vertex_1.co = self.get_reference_position( index, 0) + self.directions[index] * self.get_distance()
+            case 2:
+                vertex_1.co = self.starting_positions[index][1]
+                vertex_0.co = self.get_reference_position( index, 1) - self.directions[index] * self.get_distance()
+
+
+        bmesh.update_edit_mesh(context.active_object.data)
+
+
+    def get_distance(self ):
+        return self.distance if not self.current_affect_mode == 0 else self.distance / 2
+ 
+    
+    def get_reference_position(self, index, index_of_vert):
+        if not self.offset_distance:
+            if self.current_affect_mode == 0:
+                return self.midpoints[index]
+            else:
+                return self.starting_positions[index][index_of_vert]
+        else:
+            return self.starting_positions[index][not index_of_vert]
+
+
+    def assign_points(self):
+        self.primary_points.clear()
+        self.secondary_points.clear()
+
+
+        match self.current_affect_mode:
+            case 0:
+                self.primary_points = [pos for pos_pair in self.current_positions for pos in pos_pair]
+            case 1:
+                self.primary_points = [pos_pair[1] for pos_pair in self.current_positions]
+                self.secondary_points = [pos_pair[0] for pos_pair in self.current_positions]
+            case 2:
+                self.primary_points = [pos_pair[0] for pos_pair in self.current_positions]
+                self.secondary_points = [pos_pair[1] for pos_pair in self.current_positions]
+    
+
+    def compare_distance_to_cursor(self, context, coords_0, coords_1):
+        cursor_pos = context.scene.cursor.location
+        return v3_distance(cursor_pos, coords_0) < v3_distance(cursor_pos, coords_1)
+
+
+    def finish(self, context):
+        unregister_draw_handler()
+        unregister_points_handler()
+
+
+    def revert(self, context):
+        self.offset_distance = True
+        self.distance = 0
+        self.operate(context)
+
+        unregister_draw_handler()
+        unregister_points_handler()
+
+
+def draw_text_callback(self):
+    draw_header(self)
+
+    draw_property(
+        self,
+        f"Distance: {(self.distance * self.display_unit_scale):.2f}{self.unit_suffix}",
+        self.unit_step_hint,
+        active=self.key_no_modifiers,
+        alt_mode=self.key_shift_no_modifiers,
+        mouse_value=True,
+        input_stream=self.distance_input_stream)
+
+
+    draw_hint(
+        self,
+        "Affect [A]: {}".format(self.affect_modes[self.current_affect_mode].capitalize()),
+        "Affect ({})".format(", ".join([m.capitalize() for m in self.affect_modes])))
+    
+
+    draw_hint(
+        self,
+        "Distance [D]: {}".format('Offset' if self.offset_distance else 'Overwrite'),
+        "Overwrite, Offset")
```

Comment:
Let's use "Absolute" and "Offset" here:

```suggestion
        "Distance [D]: {}".format('Offset' if self.offset_distance else 'Absolute'),
        "Absolute, Offset")
```